src/colorizer/AnnotationData.ts

```diff
@@ -517,6 +526,91 @@ export class AnnotationData implements IAnnotationData {
     }
     return { annotationData, mismatchedTimes, mismatchedTracks, unparseableRows, invalidIds, totalRows: data.length };
   }
+  /**
+   * Returns the index of a label with the matching name and type.
+   * Returns -1 if no such label exists.
+   */
+  private findMatchingLabelIdx(name: string, type: LabelType): number {
+    for (let i = 0; i < this.labelData.length; i++) {
+      const labelData = this.labelData[i];
+      if (labelData.options.name === name && labelData.options.type === type) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Merges two annotation data objects and returns a new, resulting object,
+   * based on the merge mode. Data is deep-copied, so the original objects are
+   * not modified.
+   * @param annotationData1 The first annotation data object.
+   * @param annotationData2 The second annotation data object.
+   * @param mergeMode The merge mode to use. Can be one of:
+   * - `AnnotationMergeMode.APPEND`: Appends the labels from the second
+   *    annotation data object to the first one.
+   * - `AnnotationMergeMode.MERGE`: Merges label data for labels that match name
+   *   and type. If a label exists in both objects, the IDs are merged, with the
+   *   second object taking precedence for the values. Labels that do not match
+   *   are appended.
+   * - `AnnotationMergeMode.OVERWRITE`: Returns a copy of the second annotation
+   *   data object.
+   * @param reassignColors If true, reassigns colors on labels appended from the
+   *   second annotation data object to the default for that index. Defaults to
+   *   true.
+   * @returns A new annotation data object with the merged label data.
+   */
+  static merge(
+    annotationData1: AnnotationData,
+    annotationData2: AnnotationData,
+    mergeMode: AnnotationMergeMode,
+    reassignColors: boolean = true
+  ): AnnotationData {
+    const mergedAnnotationData = new AnnotationData();
+
+    if (mergeMode === AnnotationMergeMode.OVERWRITE) {
+      mergedAnnotationData.labelData = [...annotationData2.labelData.map(cloneLabel)];
+      mergedAnnotationData.numLabelsCreated = annotationData2.numLabelsCreated;
```

Comment:
Are we assuming that the new annotation data we're overwriting with is already the default colors, is that why we're not bothering with checking `reassignColors` in this case?

Reply:
Yes! I could probably make the change just in case, though.